From db01dfe285fbd06b1fb4790970b3941672257853 Mon Sep 17 00:00:00 2001
From: Peter Swain <pswain@cavium.com>
Date: Sat, 16 Sep 2017 13:51:05 -0700
Subject: [PATCH 299/375] mtd: nand: add Micron MT29F1G01AAADD, via
 staging/mt29f_spinand

Adds this 1Gb SPI part with internal ECC.
Still exploring ECC warning:
  nand: device found, Manufacturer ID: 0x2c, Chip ID: 0x12
  nand: Micron NAND 128MiB 3,3V 1-bit
  nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
  nand: WARNING: spi0.1: the ECC used on your system is too weak compared to the one required by the NAND chip

Signed-off-by: Peter Swain <pswain@cavium.com>
---
 drivers/mtd/nand/nand_ids.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/mtd/nand/nand_ids.c b/drivers/mtd/nand/nand_ids.c
index 2af9869a115e..f0e3e1f1f209 100644
--- a/drivers/mtd/nand/nand_ids.c
+++ b/drivers/mtd/nand/nand_ids.c
@@ -162,6 +162,20 @@ struct nand_flash_dev nand_flash_ids[] = {
 	EXTENDED_ID_NAND("NAND 64GiB 1,8V 16-bit", 0x2E, 65536, LP_OPTIONS16),
 	EXTENDED_ID_NAND("NAND 64GiB 3,3V 16-bit", 0x4E, 65536, LP_OPTIONS16),
 
+	/* Micron MT29F1G01AAADD, via staging/mt29f_spinand */
+	{
+		.name = "NAND 128MiB 3,3V 1-bit",
+		.mfr_id = NAND_MFR_MICRON,
+		.dev_id = 0x12,
+		.pagesize = 2048,
+		.chipsize = 128,
+		.erasesize = SZ_128K,
+		.options = NAND_BBT_SCAN2NDPAGE | NAND_ECC_CUSTOM_PAGE_ACCESS,
+		.id_len = 2,
+		.oobsize = 64,
+		.ecc = NAND_ECC_INFO(4, 512),
+	},
+
 	{NULL}
 };
 
-- 
2.14.1

